core/lib/spree/app_configuration.rb

```diff
@@ -169,6 +169,10 @@ class AppConfiguration < Preferences::Configuration
     #   @return [String] Email address used as +From:+ field in transactional emails.
     preference :mails_from, :string, default: 'solidus@example.com'
 
+    # @!attribute [rw] associate_user_in_authentication_extension
+    #   @return [Boolean] manually call associate_user instead of relying on current_order
+    preference :associate_user_in_authentication_extension, :boolean, default: by_version(false, "4.0.0" => true)
```

Comment:
```suggestion
    preference :associate_user_in_authentication_extension, :boolean, default: by_version(false, "3.2.0.alpha" => true)
```

We're going to remove the setting in v4.0 completely. We want it to be `false` on 3.1, but `true` on master (v3.2.0.alpha) and onward (v3.2.0). When users update v3.1 to v3.2 with `bin/rails g solidus:update`, they will keep having it to `false` as it will be overridden by the generated initializer `new_solidus_defaults.rb`. Then, they'll adapt their app (if needed) and remove the override. Once done, they'll be using `true` (default in v3.2), and their app will be ready for the future update to v4.0, where we'll completely remove all references to `:associate_user_in_authentication_extension`.

Once done this change, no error should be raised on the CI.